src/otx/engine/utils/auto_configurator.py

```diff
@@ -0,0 +1,266 @@
+"""Auto-Configurator class & util functions for OTX Auto-Configuration."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import datumaro
+from lightning.pytorch.cli import instantiate_class
+
+from otx.core.config.data import DataModuleConfig, SubsetConfig, TilerConfig
+from otx.core.data.dataset.base import LabelInfo
+from otx.core.data.module import OTXDataModule
+from otx.core.types.task import OTXTaskType
+from otx.core.utils import get_otx_root_path
+from otx.core.utils.instantiators import partial_instantiate_class
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+
+    from otx.core.model.entity.base import OTXModel
+
+
+logger = logging.getLogger()
+RECIPE_PATH = get_otx_root_path() / "recipe"
+
+DEFAULT_CONFIG_PER_TASK = {
+    OTXTaskType.MULTI_CLASS_CLS: RECIPE_PATH / "classification" / "multi_class_cls" / "otx_efficientnet_b0.yaml",
+    OTXTaskType.MULTI_LABEL_CLS: RECIPE_PATH / "classification" / "multi_label_cls" / "efficientnet_b0_light.yaml",
+    OTXTaskType.DETECTION: RECIPE_PATH / "detection" / "atss_mobilenetv2.yaml",
+    OTXTaskType.SEMANTIC_SEGMENTATION: RECIPE_PATH / "semantic_segmentation" / "litehrnet_18.yaml",
+    OTXTaskType.INSTANCE_SEGMENTATION: RECIPE_PATH / "instance_segmentation" / "maskrcnn_r50.yaml",
+    OTXTaskType.ACTION_CLASSIFICATION: RECIPE_PATH / "action" / "action_classification" / "x3d.yaml",
+    OTXTaskType.ACTION_DETECTION: RECIPE_PATH / "action" / "action_detection" / "x3d_fastrcnn.yaml",
+    OTXTaskType.VISUAL_PROMPTING: RECIPE_PATH / "visual_prompting" / "sam_tiny_vit.yaml",
+}
+
+TASK_PER_DATA_FORMAT = {
+    "imagenet_with_subset_dirs": [OTXTaskType.MULTI_CLASS_CLS],
+    "datumaro": [OTXTaskType.MULTI_LABEL_CLS],
+    "coco_instances": [OTXTaskType.DETECTION, OTXTaskType.INSTANCE_SEGMENTATION, OTXTaskType.VISUAL_PROMPTING],
+    "common_semantic_segmentation_with_subset_dirs": [OTXTaskType.SEMANTIC_SEGMENTATION],
+    "kinetics": [OTXTaskType.ACTION_CLASSIFICATION],
+    "ava": [OTXTaskType.ACTION_DETECTION],
+}
+
+
+def configure_task(data_root: str | Path | None) -> OTXTaskType | None:
```

Comment:
Why its return type can be `None`?

Reply:
Fixed this function to not accept None. - https://github.com/openvinotoolkit/training_extensions/pull/2841/commits/254e488ff916eccef718764c8ae2844239014642